mmv1/third_party/terraform/services/securitycenterv2/resource_scc_v2_project_big_query_export_config_test.go

```diff
@@ -0,0 +1,148 @@
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+)
+
+func TestAccSecurityCenterV2ProjectBigQueryExportConfig_basic(t *testing.T) {
+	t.Parallel()
+
+	randomSuffix := acctest.RandString(t, 10)
+	dataset_id := "tf_test_" + randomSuffix
+	orgID := envvar.GetTestOrgFromEnv(t)
+
+	context := map[string]interface{}{
+		"org_id":        orgID,
+		"random_suffix": randomSuffix,
+		"dataset_id":    dataset_id,
+		"dataset": fmt.Sprintf("projects/%s/datasets/%s",
+			envvar.GetTestProjectFromEnv(), dataset_id),
+		"big_query_export_id": "tf-test-export-" + randomSuffix,
+		"name": fmt.Sprintf("projects/%s/locations/global/bigQueryExports/%s",
+			envvar.GetTestProjectFromEnv(), "tf-test-export-"+randomSuffix),
+		"project": envvar.GetTestProjectFromEnv(),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"random": {},
+			"time":   {},
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSecurityCenterV2ProjectBigQueryExportConfig_basic(context),
+			},
+			{
+				ResourceName:            "google_scc_v2_project_scc_big_query_exports.default",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"update_time", "project"},
+			},
+			{
+				Config: testAccSecurityCenterV2ProjectBigQueryExportConfig_update(context),
+			},
+			{
+				ResourceName:            "google_scc_v2_project_scc_big_query_exports.default",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"update_time", "project"},
+			},
+		},
+	})
+}
+
+func testAccSecurityCenterV2ProjectBigQueryExportConfig_basic(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+
+resource "google_bigquery_dataset" "default" {
+  dataset_id                  = "%{dataset_id}"
+  friendly_name               = "test"
+  description                 = "This is a test description"
+  location                    = "US"
+  default_table_expiration_ms = 3600000
+  default_partition_expiration_ms = null
+
+  labels = {
+    env = "default"
+  }
+
+  lifecycle {
+	ignore_changes = [default_partition_expiration_ms]
+  }
+}
+
+resource "time_sleep" "wait_1_minute" {
+	depends_on = [google_bigquery_dataset.default]
+	create_duration = "3m"
+}
+
+resource "google_scc_v2_project_scc_big_query_exports" "default" {
+  name		   = "%{name}"
+  big_query_export_id    = "%{big_query_export_id}"
+  project      = "%{project}"
+  dataset      = "%{dataset}"
+  location     = "global"
+  description  = "Cloud Security Command Center Findings Big Query Export Config"
+  filter       = "state=\"ACTIVE\" AND NOT mute=\"MUTED\""
+
+  lifecycle {
+	ignore_changes = [name]
+  }
```

Comment:
this worries me. Why are we ignoring changes on Name? It doesn't appear right. Name should be output only if it is a field the user does not control it. Then this line can be removed.

Reply:
I was getting 'Plan was not empty' error because of the `name` so I have ignored it. However, I have removed that piece now.

Reply:
Yes this would make sense if the API was returning it. Terraform thought you meant to set intentional set name to empty. Thus there was a diff between the API and terraform config. With the transition to name being `output-only` there is an an understanding that this is set by the API and not controlled through terraform.